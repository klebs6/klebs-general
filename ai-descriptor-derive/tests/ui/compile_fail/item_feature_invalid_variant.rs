// ---------------- [ File: ai-descriptor-derive/tests/ui/compile_fail/item_feature_invalid_variant.rs ]
#![allow(unused_imports)]
use ai_descriptor_derive::*;
use ai_descriptor_trait::*;

#[derive(ItemFeature)]
enum PotionEffect {
    #[ai("Makes you invisible.")]
    Invisibility,
    Complex { a: String, b: i32 }, // Invalid variant type
}

fn main() {}
