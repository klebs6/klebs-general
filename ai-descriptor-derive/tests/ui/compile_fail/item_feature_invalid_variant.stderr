error: Variants must be unit variants with #[ai("...")] or single-field tuple variants wrapping a type that implements ItemFeature
  --> tests/ui/compile_fail/item_feature_invalid_variant.rs:10:5
   |
10 |     Complex { a: String, b: i32 }, // Invalid variant type
   |     ^^^^^^^
